cmd/fpga_admissionwebhook/patcher_test.go

```diff
@@ -0,0 +1,353 @@
+			},
+			interfaceID: "ce48969398f05f33946d560708be108a",
+		},
+		{
+			input: "fpga.intel.com/arria10-unknown",
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			expectedErr: true,
+		},
+		{
+			input:       "fpga.intel.com/unknown",
+			expectedErr: true,
+		},
+		{
+			input: "fpga.example.com/something",
+		},
+		{
+			input: "fpga.intel.com/arria10-nlb0",
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+			},
+			interfaceID: "ce48969398f05f33946d560708be108a",
+			afuID:       "d8424dc4a4a3c413f89e433683f9040b",
+		},
+	}
+
+	for num, tt := range tcases {
+		p := &patcher{
+			afMap:     tt.afMap,
+			regionMap: tt.regionMap,
+		}
+		interfaceID, afuID, err := p.parseResourceName(tt.input)
+		if tt.expectedErr {
+			if err != nil {
+				continue
+			} else {
+				t.Errorf("In case %d we didn't get error", num)
+			}
+		}
+		if tt.interfaceID != interfaceID || tt.afuID != afuID {
+			t.Errorf("In case %d expected (%s, %s), but got (%s, %s)", num, tt.interfaceID, tt.afuID, interfaceID, afuID)
+		}
+	}
+}
+
+func TestGetPatchOpsOrchestrated(t *testing.T) {
+	tcases := []struct {
+		name        string
+		container   corev1.Container
+		afMap       map[string]string
+		regionMap   map[string]string
+		expectedErr bool
+		expectedOps int
+	}{
+		{
+			name: "Successful handling",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"cpu": resource.MustParse("1"),
+					},
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"cpu": resource.MustParse("1"),
+					},
+				},
+			},
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+			},
+			expectedOps: 3,
+		},
+		{
+			name: "More than one FPGA in Limits",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"fpga.intel.com/arria10-nlb3": resource.MustParse("1"),
+					},
+				},
+			},
+			regionMap: map[string]string{
+				"arria10": "ce48969398f05f33946d560708be108a",
+			},
+			afMap: map[string]string{
+				"arria10-nlb0": "d8424dc4a4a3c413f89e433683f9040b",
+				"arria10-nlb3": "f7df405cbd7acf7222f144b0b93acd18",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "More than one FPGA in Requests",
+			container: corev1.Container{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"fpga.intel.com/arria10-nlb0": resource.MustParse("1"),
+						"fpga.intel.com/arria10-nlb3": resource.MustParse("1"),
+					},
```

Comment:
what will happen in case one in limits another one in requests ?

Reply:
There's no check for this in the webhook. I expect the API server takes care of that since it returns
```
spec.containers[0].resources.limits: Required value: Limit must be set for non overcommitable resources
```
in such cases before consulting the webhook.

Reply:
what I meant was something like 
```!yaml
- requests:
   fpga.intel.com/arria10-nlb0: 1
- limits:
   fpga.intel.com/arria10-nlb3: 1
```